src/gg.regression.unity.bots/Runtime/Scripts/RGServiceManager.cs

```diff
@@ -498,6 +498,119 @@ public async Task UploadLogs(long botInstanceId, string filePath, Action onSucce
                 onFailure();
             }
         }
+        
+        public async Task CreateGameplaySession(DateTime startTime, DateTime endTime, long numTicks, Action<RGGameplaySession> onSuccess, Action onFailure)
+        {
+            if (LoadAuth())
+            {
+                RGGameplaySessionCreateRequest request = new RGGameplaySessionCreateRequest(startTime, endTime, numTicks);
+                await SendWebRequest(
+                    uri: $"{GetRgServiceBaseUri()}/gameplay-session",
+                    method: "POST",
+                    payload: JsonConvert.SerializeObject(request),
+                    onSuccess: (s) =>
+                    {
+                        RGGameplaySession response = JsonUtility.FromJson<RGGameplaySession>(s);
+                        RGDebug.LogDebug(
+                            $"RGService CreateGameplaySession response received");
+                        onSuccess.Invoke(response);
+                    },
+                    onFailure: (f) =>
+                    {
+                        RGDebug.LogWarning($"Failed to create gameplay session: {f}");
+                        onFailure.Invoke();
+                    }
+                );
+            }
+            else
+            {
+                onFailure();
+            }
+        }
+        
+        public async Task UploadGameplaySessionData(long gameplaySessionId, string zipPath, Action onSuccess, Action onFailure)
```

Comment:
[Non-Blocking] UploadGameplaySessionData and UploadGameplaySessionScreenshots are almost identitical. We could create a single method with one extra `contentType` param:
```
private async Task UploadData(string uri, string filePath, string contentType, Action onSuccess, Action onFailure)
{
    if (LoadAuth())
    {
        await SendWebFileUploadRequest(
            uri: uri,
            method: "POST",
            filePath: filePath,
            contentType: contentType,
            onSuccess: (s) =>
            {
                RGDebug.LogDebug($"Response received for {uri}");
                onSuccess.Invoke();
            },
            onFailure: (f) =>
            {
                RGDebug.LogWarning($"Failed to upload for {uri}: {f}");
                onFailure.Invoke();
            }
        );
    }
    else
    {
        onFailure();
    }
}

public async Task UploadGameplaySessionData(long gameplaySessionId, string zipPath, Action onSuccess, Action onFailure)
{
    await UploadData(
        $"{GetRgServiceBaseUri()}/gameplay-session/{gameplaySessionId}/data",
        zipPath,
        "application/zip",
        onSuccess,
        onFailure);
}

// same thing for screenshots
```

Reply:
I think this could be fine, but also since we are only doing this in two spots, I feel that an extra function may just add more redirection for readers of the code.